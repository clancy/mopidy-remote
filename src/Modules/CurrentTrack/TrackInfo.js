/* @flow */
import React, { Component, PropTypes } from 'react';
import {
  View,
  StyleSheet,
  Text
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const iconStyle = enabled => {
  return {
      'color' : enabled ? 'green' : 'white'
  };
};

const TrackInfo = React.createClass({
  propTypes: {
    inLibrary: PropTypes.bool.isRequired,
    trackName: PropTypes.string,
    artists: PropTypes.arrayOf(React.PropTypes.shape({
      name: PropTypes.string.isRequired
    })).isRequired
  },
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.iconContainer}>
          <Icon style={iconStyle(this.props.inLibrary)} name='done' size={35} />
        </View>
        <View style={styles.trackTextView}>
          <Text style={styles.trackNameText}>{this.props.trackName}</Text>
          <Text style={styles.artistText}>{this.props.artists.map(a => a.name).join(', ')}</Text>
        </View>
        <View style={styles.iconContainer}>
          <Icon style={iconStyle()}  name='more-vert' size={35} />
        </View>
      </View>
    );
  }
});

let styles = StyleSheet.create({
  container: {
    flex: .5,
    backgroundColor: 'black',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  iconContainer: {
    flex: 15,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  trackNameText: {
    color: 'white',
    fontWeight: 'bold'
  },
  artistText: {
    color: 'white'
  },
  trackTextView: {
    flexDirection: 'column',
    alignItems: 'center',
    flex: 70
  }
});

export default TrackInfo;
