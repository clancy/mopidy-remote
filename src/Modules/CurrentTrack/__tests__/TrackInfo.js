import 'react-native';
import React from 'react';
import TrackInfo from '../TrackInfo';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('TrackInfo tests', () => {
  test('renders TrackInfo correctly with with inLibrary false', () => {
    const tree = renderer.create(
      <TrackInfo
        inLibrary={false}
        trackName={"Harry"}
        artists={[{name:"Sally"}]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders TrackInfo correctly with with inLibrary true', () => {
    const tree = renderer.create(
      <TrackInfo
        inLibrary={true}
        trackName={"Harry"}
        artists={[{name:"Sally"}]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders TrackInfo correctly with with multiple artists', () => {
    const tree = renderer.create(
      <TrackInfo
        inLibrary={true}
        trackName={"Harry"}
        artists={[{name:"Sally"}, {name:"Fred"}]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
