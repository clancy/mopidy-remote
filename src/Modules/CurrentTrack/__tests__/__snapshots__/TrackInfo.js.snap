// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackInfo tests renders TrackInfo correctly with with inLibrary false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 15,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 70,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Harry
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Sally
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 15,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`TrackInfo tests renders TrackInfo correctly with with inLibrary true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 15,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "green",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 70,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Harry
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Sally
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 15,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`TrackInfo tests renders TrackInfo correctly with with multiple artists 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 15,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "green",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 70,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Harry
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Sally, Fred
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 15,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
